returnn/datasets/hdf.py

```diff
@@ -1287,27 +1297,35 @@ def dump_from_dataset(self, dataset, epoch=1, start_seq=0, end_seq=float("inf"),
       data_keys.remove("raw")
     data_target_keys = [key for key in dataset.get_target_list() if key in data_keys]
     data_input_keys = [key for key in data_keys if key not in data_target_keys]
-    assert len(data_input_keys) > 0
-    if len(data_input_keys) > 1:
-      if "data" in data_input_keys:
-        default_data_input_key = "data"
+    default_data_input_key = None
+    progress_bar_data_key = None
+    if data_input_keys:
+      if len(data_input_keys) > 1:
+        if "data" in data_input_keys:
+          default_data_input_key = "data"
+        else:
+          raise Exception("not sure which input data key to use from %r" % (data_input_keys,))
       else:
-        raise Exception("not sure which input data key to use from %r" % (data_input_keys,))
-    else:
-      default_data_input_key = data_input_keys[0]
+        default_data_input_key = data_input_keys[0]
+      progress_bar_data_key = default_data_input_key
     print("Using input data key:", default_data_input_key)
+    default_data_target_key = None
```

Comment:
Do we actually still need `default_data_target_key`?

Reply:
Only to create this `attr_numLabels = 'numLabels'` field in the HDF. I don't know if anyone uses it anymore though, it has been replaced by `'targets/size'`, see line 163.

Reply:
Yea, since a long time. I think it would be safe to remove.

Reply:
What about `inputPattSize`? Shall I remove both in one go?

Reply:
Ah no, for sparse "data" it seems to be used.